### DISCUSSION TITLE
Documentation or examples for Silk.NET.OpenCL?

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/Silk.NET#654] [Q&A] [answered]
I have been unable to find any documentation or examples for using Silk.NET.OpenCL, does any sample code exists? It would be nice to have basic initialization and running some kernel if anyone can provide such an example.

**Comment by Perksey** [comment, marked as answer]
We currently do not have any official examples for using Silk.NET.OpenCL. It does mirror the C API quite closely, only that all strongly typed handles afaik have been replaced with “nint”.

**Reply by justinmchase** [reply]
Do you have a link to the C API documentation?

Even a small example here would be super helpful.

**Reply by Beyley** [reply]
> Do you have a link to the C API documentation?
> 
> Even a small example here would be super helpful.

you should be able to find it relatively easily by googling "OpenCL specification", and clicking on one of the khronos links
